node/node_test.go

```diff
@@ -738,6 +744,160 @@ func TestCreateTransaction_OkValsShouldWork(t *testing.T) {
+	)
+
+	nonce := uint64(0)
+	value := new(big.Int).SetInt64(10)
+	receiver := "rcv"
+	sender := "snd"
+	gasPrice := uint64(10)
+	gasLimit := uint64(20)
+	txData := []byte("-")
+	signature := "617eff4f"
+
+	options := uint32(2 | 16777216)
+	tx, txHash, err := n.CreateTransaction(nonce, value.String(), receiver, sender, gasPrice, gasLimit, txData, signature, string(chainID), version, options)
+	assert.NotNil(t, tx)
+	assert.Equal(t, expectedHash, txHash)
+	assert.Nil(t, err)
+	assert.Equal(t, nonce, tx.Nonce)
+	assert.Equal(t, value, tx.Value)
+	assert.True(t, bytes.Equal([]byte(receiver), tx.RcvAddr))
+
+	err = n.ValidateTransaction(tx)
+	assert.Equal(t, process.ErrInvalidTransactionVersion, err)
+}
+
+func TestCreateTransaction_TxSignedWithHashNoEnabledShouldErr(t *testing.T) {
+	t.Parallel()
+
+	expectedHash := []byte("expected hash")
+	crtShardID := uint32(1)
+	chainID := []byte("chain ID")
+	version := uint32(1)
+	n, _ := node.NewNode(
+		node.WithInternalMarshalizer(getMarshalizer(), testSizeCheckDelta),
+		node.WithVmMarshalizer(getMarshalizer()),
+		node.WithTxSignMarshalizer(getMarshalizer()),
+		node.WithHasher(
+			mock.HasherMock{
+				ComputeCalled: func(s string) []byte {
+					return expectedHash
+				},
+			},
+		),
+		node.WithAddressPubkeyConverter(
+			&mock.PubkeyConverterStub{
+				DecodeCalled: func(hexAddress string) ([]byte, error) {
+					return []byte(hexAddress), nil
+				},
+			}),
+		node.WithAccountsAdapter(&mock.AccountsStub{}),
+		node.WithShardCoordinator(&mock.ShardCoordinatorMock{
+			ComputeIdCalled: func(i []byte) uint32 {
+				return crtShardID
+			},
+			SelfShardId: crtShardID,
+		}),
+		node.WithWhiteListHandler(&mock.WhiteListHandlerStub{}),
+		node.WithWhiteListHandlerVerified(&mock.WhiteListHandlerStub{
+			IsWhiteListedCalled: func(interceptedData process.InterceptedData) bool {
+				return false
+			},
+		}),
+		node.WithKeyGenForAccounts(&mock.KeyGenMock{
+			PublicKeyFromByteArrayMock: func(b []byte) (crypto.PublicKey, error) {
+				return nil, nil
+			},
+		}),
+		node.WithTxSingleSigner(&mock.SingleSignerMock{}),
+		node.WithTxFeeHandler(&mock.FeeHandlerStub{
+			CheckValidityTxValuesCalled: func(tx process.TransactionWithFeeHandler) error {
+				return nil
+			},
+		}),
+		node.WithChainID(chainID),
+		node.WithMinTransactionVersion(version),
+		node.WithEpochStartTrigger(&mock.EpochStartTriggerStub{
+			EpochCalled: func() uint32 {
+				return 1
+			},
+		}),
+		node.WithEnableSignTxWithHashEpoch(2),
+		node.WithTxSignHasher(&mock.HasherMock{}),
+	)
+
+	nonce := uint64(0)
+	value := new(big.Int).SetInt64(10)
+	receiver := "rcv"
+	sender := "snd"
+	gasPrice := uint64(10)
+	gasLimit := uint64(20)
+	txData := []byte("-")
+	signature := "617eff4f"
+
+	options := uint32(1 | 16777216)
+	tx, txHash, err := n.CreateTransaction(nonce, value.String(), receiver, sender, gasPrice, gasLimit, txData, signature, string(chainID), version+1, options)
+	assert.NotNil(t, tx)
```

Comment:
in this test and in the previous one, no need for all the asserts after `CreateTransaction` because they are tested in another unit test. keep just the tx validation asserts

Reply:
done.